pages/api/drive/private/download.ts

```diff
@@ -0,0 +1,33 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import {
+  getDriveInstance,
+  getDriveFile,
+  checkIfLoggedIn,
+} from '../../../../utils/driveFiles'
+
+const drive = getDriveInstance(process.env.GOOGLE_PRIVATE_CREDS)
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  try {
+    //check if logged in
+    const isLoggedIn = await checkIfLoggedIn(req.headers.cookie)
+    if (!isLoggedIn) {
+      res.status(403).json({ error: 'User not logged in' })
+    }
```

Comment:
You could move the login check to a [middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware)